File: cv_py/webcam.py
import cv2
import sys
import time

try:
    # Camera
    camera = cv2.VideoCapture(0)

    # Cascade file for face detection
    cascade_path = "/usr/local/Cellar/opencv/4.5.1_2/share/opencv4/lbpcascades/lbpcascade_frontalface.xml"
    cascade = cv2.CascadeClassifier(cascade_path)

    # Frame processing loop
    start_time = time.time()
    frame_cnt = 0
    while True:
        # Get frame
        ret_val, frame = camera.read()

        # Convert to gray scale
        gray = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)

        # Detect faces
        facerect = cascade.detectMultiScale(gray)

        # Draw rect to frame
        for rect in facerect:
            cv2.rectangle(frame, tuple(rect[0:2]),tuple(rect[0:2]+rect[2:4]), (0,0,255), thickness=3)

        # Display
        cv2.imshow('Face Detected', frame)
        if cv2.waitKey(1) == ord('q'):
            break

        frame_cnt += 1

    end_time = time.time()
    fps = frame_cnt / int(end_time - start_time)
    print("FPS=" + "{:.1f}".format(fps))

except Exception as e:
    print(str(e))

File: cv_cpp/webcam.cpp
#include <opencv2/opencv.hpp>
#include <time.h>

using namespace std;

int main (int argc, char **argv){
    try {
        // Camera
        cv::VideoCapture camera(0);
        if (!camera.isOpened()) {
            cerr << "Unable to connect to camera" << endl;
            return 1;
        }
    
        // Cascade file for face detection
        const std::string cascade_name = "/usr/local/share/opencv4/lbpcascades/lbpcascade_frontalface.xml";
        cv::CascadeClassifier cascade;
        cascade.load(cascade_name);
    
        // Frame processing loop
        time_t start_time = time(NULL);
        double frame_cnt = 0;
        while (cv::waitKey(1) != 'q') {
            // Get frame
            cv::Mat frame, gray;
            camera >> frame;
            if (frame.empty() == true) break;

            // Convert to gray scale
            cv::cvtColor(frame, gray, cv::COLOR_BGRA2GRAY);
    
            // Detect faces
            std::vector<cv::Rect> faces;
            cascade.detectMultiScale(gray, faces);
    
            // Draw ract to frame
            for(std::vector<cv::Rect>::iterator it=faces.begin(); it!=faces.end(); it++){
                cv::rectangle(frame, *it, CV_RGB(255,0,0), 3);
            }
    
            // Display
            cv::imshow("Face Detected", frame);

            frame_cnt++;
        }
        // FPS
        time_t end_time = time(NULL);
        double fps = frame_cnt / (end_time - start_time);
        cout << "FPS=" << std::setprecision(2) << fps << endl;
    } catch(exception& e) {
        cout << e.what() << endl;
    }

    return 0;
}
